clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -729,6 +729,23 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
     private class OffsetFetchResponseHandler extends CoordinatorResponseHandler<OffsetFetchResponse, Map<TopicPartition, OffsetAndMetadata>> {
         @Override
         public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartition, OffsetAndMetadata>> future) {
+            if (response.hasError()) {
+                Errors error = Errors.forCode(response.errorCode());
+                log.debug("Offset fetch for group {} failed: {}", groupId, error.message());
+
+                if (error == Errors.GROUP_LOAD_IN_PROGRESS) {
```

Comment:
So if we want to just check the top level error for any error in the response for partition level errors we lose the specific partitions that are erroneous. Also, if there are different partition level error types present we'll report only one. This just limits the error reporting but shouldn't affect the functionality. I hope I did not misunderstand your point.